x-pack/plugins/enterprise_search/public/applications/app_search/routes.ts

```diff
@@ -10,11 +10,15 @@ import { docLinks } from '../shared/doc_links';
 export const DOCS_PREFIX = docLinks.appSearchBase;
 
 export const ROOT_PATH = '/';
+export const NOT_FOUND_PATH = '/404';
```

Comment:
This addition confuses me, I don't think we're using it anywhere 🤔  Can we nuke it?

```suggestion
```

Reply:
Sorry, I can remove for now. It is needed in a future PR:

```JavaScript
      try {
        const response = await http.get(route);
        actions.setRoleMappingData(response);
      } catch (e) {
        if (e.status === 404) {
          navigateToUrl(NOT_FOUND_PATH);
        }
        flashAPIErrors(e);
      }
```
FWIW, this is currently broken in `ent-search`. If you navigate to a non-existent Role mapping, you just land on the listing page with no feedback

Reply:
I'd also say don't bother checking the status for 404. If the role mapping fails to load for whatever reason (even a 500), the whole page will be wonky and have potential errors, so we should always opt to redirect back to the main page.